### PR TITLE
Use GitHub Actions for CI

### PR DESCRIPTION
**Pull request by sagudev**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        brew install python autoconf@2.13 ccache llvm yasm
+    - name: Install deps on linux
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        sudo apt install autoconf2.13 gcc-7 g++-7 ccache llvm -y
+    - name: Install deps on windows
+      if: startsWith(matrix.os, 'windows')
+      working-directory: "C:\\"
+      run: |
+        cp D:\a C:\ -Recurse
```
Comment (sagudev) [inline .github/workflows/rust.yml:35]: For some wired reason we need to have src on C drive.
```
 error: failed to run custom build command for `mozjs_sys v0.68.2 (https://github.com/servo/mozjs?rev=c6c7b5319975a8f0465ce8c17329e74be4cb80b1#c6c7b531)`

Caused by:
  process didn't exit successfully: `D:\a\rust-mozjs\rust-mozjs\target\debug\build\mozjs_sys-5c475a09377ae625\build-script-build` (exit code: 101)
  --- stdout
  cargo:outdir=D:\a\rust-mozjs\rust-mozjs\target\debug\build\mozjs_sys-4dd1a798d5c68344\out\build
  [[ 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/configure -ot 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/configure.in ]] && touch 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/configure || true
  [[ 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/old-configure -ot 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/old-configure.in ]] && touch 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/old-configure || true
  ! [[ 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/configure.in -ot 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/configure ]] && touch 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/configure || true
  ! [[ 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/old-configure.in -ot 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/old-configure ]] && touch 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/old-configure || true
  if [[ 'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/configure -nt config.status ]] ; then \
    PYTHON="C:\mozilla-build\python\python.exe" MOZ_TOOLS="/" \
    CC="clang-cl.exe" CFLAGS="-MD" \
    CPP="" CPPFLAGS="-IC:/Users/runneradmin/.cargo/registry/src/github.com-1ecc6299db9ec823/encoding_c_mem-0.2.6/include " \
    CXX="clang-cl.exe" CXXFLAGS="-MD" \
    AS="" AR="" \
    STLPORT_LIBS="" \
    'C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs'/js/src/configure --disable-jemalloc --disable-js-shell --disable-tests --disable-export-js --disable-shared-js --build-backends=RecursiveMake --enable-debug --disable-optimize --enable-gczeal --target=x86_64-pc-mingw32 --host=x86_64-pc-mingw32 || (cat config.log && exit 1) ; \
  fi
  Creating Python 3 environment
  ['C:/Users/runneradmin/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/c6c7b53/mozjs\\third_party\\python\\virtualenv', 'C:\\Users\\runneradmin\\.cargo\\git\\checkouts\\mozjs-fa11ffc7d4f1cc2d\\c6c7b53\\mozjs\\third_party\\python\\virtualenv', 'C:\\mozilla-build\\python3\\python37.zip', 'C:\\mozilla-build\\python3\\DLLs', 'C:\\mozilla-build\\python3\\lib', 'C:\\mozilla-build\\python3']
  created virtual environment CPython3.7.4.final.0-64 in 1302ms
    creator CPython3Windows(dest=D:\a\rust-mozjs\rust-mozjs\target\debug\build\mozjs_sys-4dd1a798d5c68344\out\build\_virtualenvs\init_py3, clear=False, global=False)
    seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=C:\Users\runneradmin\AppData\Local\pypa\virtualenv)
      added seed packages: pip==20.2.2, setuptools==49.6.0, wheel==0.35.1
    activators BashActivator,BatchActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
  Traceback (most recent call last):
    File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 771, in <module>
      manager.populate()
    File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 463, in populate
      handle_package(package)
    File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 381, in handle_package
      submanager.populate(sitecustomize=sitecustomize)
    File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 463, in populate
      handle_package(package)
    File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 398, in handle_package
      f.write("%s\n" % os.path.relpath(path, python_lib))
    File "C:\mozilla-build\python3\lib\ntpath.py", line 562, in relpath
      path_drive, start_drive))
  ValueError: path is on mount 'C:', start on mount 'D:'
  DEBUG: python3: running with pid 2828
  DEBUG: python3: sys.executable: 'c:\\mozilla-build\\python3\\python3.exe'
  DEBUG: python3: executable from configuration: 'C:\\mozilla-build\\python3\\python3.exe'
  DEBUG: python3: found executable: 'C:/mozilla-build/python3/python3.exe'
  INFO: Creating Python 3 environment
  INFO: ['C:/Users/runneradmin/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/c6c7b53/mozjs\\third_party\\python\\virtualenv', 'C:\\Users\\runneradmin\\.cargo\\git\\checkouts\\mozjs-fa11ffc7d4f1cc2d\\c6c7b53\\mozjs\\third_party\\python\\virtualenv', 'C:\\mozilla-build\\python3\\python37.zip', 'C:\\mozilla-build\\python3\\DLLs', 'C:\\mozilla-build\\python3\\lib', 'C:\\mozilla-build\\python3']
  INFO: created virtual environment CPython3.7.4.final.0-64 in 1302ms
  INFO:   creator CPython3Windows(dest=D:\a\rust-mozjs\rust-mozjs\target\debug\build\mozjs_sys-4dd1a798d5c68344\out\build\_virtualenvs\init_py3, clear=False, global=False)
  INFO:   seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=C:\Users\runneradmin\AppData\Local\pypa\virtualenv)
  INFO:     added seed packages: pip==20.2.2, setuptools==49.6.0, wheel==0.35.1
  INFO:   activators BashActivator,BatchActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
  INFO: Traceback (most recent call last):
  INFO:   File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 771, in <module>
  INFO:     manager.populate()
  INFO:   File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 463, in populate
  INFO:     handle_package(package)
  INFO:   File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 381, in handle_package
  INFO:     submanager.populate(sitecustomize=sitecustomize)
  INFO:   File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 463, in populate
  INFO:     handle_package(package)
  INFO:   File "C:\Users\runneradmin\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\c6c7b53\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 398, in handle_package
  INFO:     f.write("%s\n" % os.path.relpath(path, python_lib))
  INFO:   File "C:\mozilla-build\python3\lib\ntpath.py", line 562, in relpath
  INFO:     path_drive, start_drive))
  INFO: ValueError: path is on mount 'C:', start on mount 'D:'
```

Comment (jdm) [reply .github/workflows/rust.yml:35]: I had to deal with this in https://github.com/servo/servo/pull/27865. you can copy the CARGO_HOME solution I found there.

Comment (jdm) [reply .github/workflows/rust.yml]: You will need github.workspace instead of env.GITHUB_WORKSPACE

Comment (sagudev) [reply .github/workflows/rust.yml]: I meant to look at it today, but you already take care of it so ...

**Pull request merged**